spaghetti/network.py

```diff
@@ -2851,18 +2826,22 @@ def _reassign(attr, cid):
 
     # provide warning (for now) if the network contains a point pattern
     if getattr(net, "pointpatterns"):
-        msg = "There is a least one point pattern associated with the network."
-        msg += " Component extraction should be performed prior to snapping"
-        msg += " point patterns to the network object; failing to do so may"
-        msg += " lead to unexpected results."
-        warnings.warn(msg)
+        msg = (
+            "There is a least one point pattern associated with the network."
+            " Component extraction should be performed prior to snapping"
+            " point patterns to the network object; failing to do so may"
+            " lead to unexpected results."
+        )
+        warnings.warn(msg, stacklevel=1)
```

Comment:
Same goes for the level here since  `extract_compontent()` is directly called by the user.

Reply:
Then I believe it should be 2. You should show the call to the function that eventually triggered the warning. If I wrapped this in some other function, I would want to know that the warning comes from calling `extract_compontent()`.

Reply:
done in [e2026eb](https://github.com/pysal/spaghetti/pull/721/commits/e2026eb6b017431cc6d473654d4b52d52ba29feb)